mantidproject/mantid
# Indirect Data Analysis I(Q, t) Fit - IqtFitMultiple algorithm

Fixes #15094

The FuryFitMult function from IndirectDataAnalysis.py has been converted into a mantid algorithm.

Related issues:
- A couple of issues have been noticed while performing this change and I have chosen to resolve these in separate issues as they are not within the scope of this issue:
- #15342
- #15343  
# To Test
- Ensure that unit tests, doc tests and system tests run and pass on the build servers
- The data for testing has been added to the unit test folder (you may need to re-run CMake and rebuild the StandardTestData package in order to access it.
# Functional testing (script)
- Run the following script in the Mantid Plot scripting window

```
#Load in iqt data
input_ws = Load(Filename='iris26176_graphite002_iqt.nxs')
function = r'name=LinearBackground,A0=0.027668,A1=0,ties=(A1=0);name=UserFunction,Formula=Intensity*exp(-(x/Tau)^Beta),Intensity=0.972332,Tau=0.0247558,Beta=1;ties=(f1.Intensity=1-f0.A0)'

#run IqtFitMultiple
result, params, fit_group_Workspaces = IqtFitMultiple(InputWorkspace=input_ws, Function=function, FitType='1S_s', SpecMin=0, SpecMax=16, StartX=0, EndX=0.2)
```
- Ensure the algorithm completes without failure or error
- This will produce 6 workspaces in the ADS
- Ignore the `__TMP` workspace, this is being resolved in a separate issue
- Ensure that the fit_group_Workspaces WorkspaceGroup has 17 entries (0-16)
- Ensure there are 2 table workspaces:
  - params
  - fit_group_NomalisedCovarianceMatrix
- Ensure there are 2 Matrix worksapces (Excluding the `__TMP`):
  - input_ws - the original data input to the algorithm
  - result
# Functional testing (GUI)
- Open `I(Q, t) Fit` (Interfaces > Indirect > Data Analysis > `I(Q, t) Fit`)
- Input file is `iris26176_graphite002_iqt.nxs` 
- Change Fit Type to `1 Stretched Exponential`
- Check the `Constrain Beta Over All Q` check box
- Change `EndX` to 0.2
- Change `Plot Output` to `All`
- Check `Save Result` - change your default save directory to somewhere you don't mind a file being saved to 
  - You can change this by clicking `Manage Directories` and the `Default Save Directory` field is at the bottom of the window
- Click `Run`
- Ensure you get 4 plots (Beta, Tau, Intensity, f0.A0) 
- Ensure the mini plot is updated with the Fit and Diff and the Plot Guess is no longer present
- Ensure that the `iris26176_graphite002_Iqt_1Smult_s0_to_16_Result` file has been saved to your default save directory

**[Release Notes](http://www.mantidproject.org/Release_Notes_3_7_Indirect_Inelastic#Algorithms)**
